qiskit_optimization/translators/docplex_mp.py

```diff
@@ -119,23 +120,172 @@ def to_docplex_mp(quadratic_program: "QuadraticProgram") -> Model:
+            x_lb *= val
+            x_ub *= val
+            linear_lb += min(x_lb, x_ub)
+            linear_ub += max(x_lb, x_ub)
+        return linear_lb, linear_ub
+
+    @classmethod
+    def _indicator_constraints(
+        cls,
+        var_names: Dict[Var, str],
+        var_bounds: Dict[str, Tuple[float, float]],
+        constraint: IndicatorConstraint,
+        indicator_big_m: Optional[float] = None,
+    ):
+        name = constraint.name
+        binary_var = constraint.binary_var
+        active_value = constraint.active_value
+        linear_constraint = constraint.linear_constraint
+        linear, sense, rhs = _FromDocplexMp._linear_constraint(var_names, linear_constraint)
+        linear_lb, linear_ub = _FromDocplexMp._linear_bounds(var_bounds, linear)
+        if sense == "<=":
+            big_m = indicator_big_m or max(0.0, linear_ub - rhs)
+            if active_value:
+                linear[binary_var.name] = big_m
+                rhs += big_m
+            else:
+                linear[binary_var.name] = -big_m
+            return [(linear, sense, rhs, name)]
+        elif sense == ">=":
+            big_m = indicator_big_m or max(0.0, rhs - linear_lb)
+            if active_value:
+                linear[binary_var.name] = -big_m
+                rhs -= big_m
+            else:
+                linear[binary_var.name] = big_m
+            return [(linear, sense, rhs, name)]
+        elif sense == "==":
+            # for equality constraints, add both GE and LE constraints.
+            # linear2, rhs2, and big_m2 are for the GE constraint.
+            linear2 = linear.copy()
+            rhs2 = rhs
+            big_m = indicator_big_m or max(0.0, linear_ub - rhs)
+            big_m2 = indicator_big_m or max(0.0, rhs - linear_lb)
+            if active_value:
+                linear[binary_var.name] = big_m
+                rhs += big_m
+                linear2[binary_var.name] = -big_m2
+                rhs2 -= big_m2
+            else:
+                linear[binary_var.name] = -big_m
+                linear2[binary_var.name] = big_m2
+            return [(linear, "<=", rhs, name + "_LE"), (linear2, ">=", rhs2, name + "_GE")]
+        else:
+            raise QiskitOptimizationError(
+                f"Internal error: invalid sense of indicator constraint: {sense}"
+            )
 
 
-def from_docplex_mp(model: Model) -> "QuadraticProgram":
+def from_docplex_mp(model: Model, indicator_big_m: Optional[float] = None) -> "QuadraticProgram":
     """Translate a docplex.mp model into a quadratic program.
 
     Note that this supports only basic functions of docplex as follows:
     - quadratic objective function
-    - linear / quadratic constraints
+    - linear / quadratic / indicator constraints
     - binary / integer / continuous variables
 
     Args:
         model: The docplex.mp model to be loaded.
+        indicator_big_m: The big-M value for indicator constraints.
```

Comment:
Is this formulation well enough known that it does not require anything further e.g. an example of an indicator contraint in the docstring here being converted so it can be seen where M comes into play.

Reply:
Yes, the big-M formulation is a well known technique. I slightly revised the text to clarify it.